src/tests/JIT/opt/OptimizeBools/optboolsreturn.cs

```diff
@@ -0,0 +1,91 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+// unit test for boolean optimization
+
+using System;
+using System.Runtime.CompilerServices;
+
+public class CBoolTest
+{
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    private static bool AreZero(int x, int y)
+    {
+        return (x == 0 && y == 0);
+    }
+
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    private static bool AreNull(object x, object y)
+    {
+        return (x == null && y == null);
+    }
+
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    private static bool AreZero2(int x, int y)
+    {
+        return x == 0 && y == 0 && BitConverter.IsLittleEndian;
+    }
+
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    private static bool AreZero3(int x, int y, int z)
+    {
+        return x == 0 && y == 0 && z == 0;
+    }
+
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    private static bool AreZero4(int x, int y, int z, int w)
+    {
+        return (x == 0 && y == 0 && z == 0 && w == 0);
+    }
+
+    // Cases that skip optimization
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    private static bool AreOne(int x, int y)
+    {
+        return (x == 1 && y == 1);
+    }
+
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    private static bool IsEitherZero(int x, int y)
+    {
+        return (x == 0 || y == 0);
+    }
+
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    private static bool IsEitherOne(int x, int y)
+    {
+        return (x == 1 || y == 1);
+    }
+
+    public static int Main()
+    {
+        try
+        {
+            // Optimize boolean
+
+            AreZero(0, 0);
```

Comment:
This test is only checking for a thrown exception on failure. I don't think you need to test for that at all; we wouldn't expect an exception in any case.

However, the function return values need to be checked for correctness.

Reply:
Updated.